### PR TITLE
feat(lint): support linting multiple charts

### LINKED ISSUE #836: lint a chart repository
I'd like for `helm lint` to be able to lint all charts in a given path.
Something like ... `helm lint --repo` might take a given path or the current path and unpack and lint every file that ends with .tgz while ignoring the index.yaml file (if present).

### PR DESCRIPTION
**Pull request by prydonius**
closes #836 

Example:
screen shot 2016-06-30 at 7 08 57 pm

### CODE REVIEW

File: cmd/helm/lint.go [hunk 1]
```diff
@@ -51,19 +51,33 @@ func init() {
 	RootCommand.AddCommand(lintCommand)
 }
 
-var errLintNoChart = errors.New("no chart found for linting (missing Chart.yaml)")
-var errLintFailed = errors.New("lint failed")
+var errLintNoChart = errors.New("No chart found for linting (missing Chart.yaml)")
```
Comment (technosophos) [inline cmd/helm/lint.go]: Go's format checker will complain about this. Try running `make test-style`.

Comment (prydonius) [inline cmd/helm/lint.go]: `gofmt` seems fine, `golint` reported a few issues though so I'll go through and fix those. Though, this line in particular didn't have any issues - are you seeing something different?

Comment (technosophos) [inline cmd/helm/lint.go]: Oh, usually it says something about not starting an error with a capitalized word.

Comment (prydonius) [inline cmd/helm/lint.go]: Hmm, you're right - looking it up and according to https://github.com/golang/go/wiki/CodeReviewComments#error-strings it should warn about it...
Shall I change it back?

**Comment by michelleN** [comment]
```
==> Linting /var/folders/yj/tmg8tvp93sx3wv8wrtc11qq40000gn/T/helm-lint843051169/alpine
Lint OK

==> Linting /var/folders/yj/tmg8tvp93sx3wv8wrtc11qq40000gn/T/helm-lint815764364/nginx
Lint OK

==> Linting /var/folders/yj/tmg8tvp93sx3wv8wrtc11qq40000gn/T/helm-lint986293627/nginx
Lint OK

```
That was my output from the linter. I really like the "==>" addition. Also it's awesome that I can now lint multiple charts using a glob, so thank you very much. The only thing I'm a little confused by is where it tells me what is being linted. Here it's showing the temp directory which is created if you're linting a packaged chart. That information is not super helpful because 1) it gets deleted after the lint run of that chart and 2) because it doesn't show the version of the chart that is being linted. 

If I have a problem with one of the alpine charts, I can't tell which one. For example:
```
helm [feat/836-lint-multiple-charts*]$ bin/helm lint testcharts/*.tgz
==> Linting /var/folders/yj/tmg8tvp93sx3wv8wrtc11qq40000gn/T/helm-lint740954906/alpine
[WARNING] templates/alpine-pod.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Release.Service}} -> : {{.Release.Service | quote }}

==> Linting /var/folders/yj/tmg8tvp93sx3wv8wrtc11qq40000gn/T/helm-lint465061809/alpine
Lint OK

==> Linting /var/folders/yj/tmg8tvp93sx3wv8wrtc11qq40000gn/T/helm-lint295968092/nginx
Lint OK

==> Linting /var/folders/yj/tmg8tvp93sx3wv8wrtc11qq40000gn/T/helm-lint668487435/nginx
Lint OK

4 chart(s) linted, no failures
```

@prydonius do you think we could change this to show something like the full pack of the package chart like "==> Linting /Users/michellenoorali/gocode/src/k8s.io/helm/testcharts/alpine-0.1.0.tgz" ? or do you have any thoughts about it?

**Comment by prydonius** [comment]
@michelleN you're right, it isn't really a helpful output when working with packaged charts. For unpackaged charts, we just display the relative dir, I suggest we do the same with packaged charts. So you're output would look more like:
```
==> Linting testcharts/alpine.tgz
...
```

So, the same as your suggestion, but with a relative path. What do you think?

**Comment by michelleN** [comment]
👍 relative path sounds good! @prydonius thank you

**Comment by michelleN** [comment]
here is something I just ran into as well. `testcharts/` doesn't have any charts in it, so the output should be `Error: 0 chart(s) linted, 0 chart(s) failed`
```
helm [feat/836-lint-multiple-charts]$ bin/helm lint testcharts/*.tgz
open testcharts/*.tgz: no such file or directory

Error: 1 chart(s) linted, 1 chart(s) failed
```

**Comment by michelleN** [comment]
This looks great: 
```
==> Linting testcharts/alpine-0.1.0.tgz
[WARNING] templates/alpine-pod.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Release.Service}} -> : {{.Release.Service | quote }}

1 chart(s) linted, no failures
```

Thanks!

**Comment by michelleN** [comment]
```
# k8s.io/helm/cmd/helm
cmd/helm/lint_test.go:29: multiple-value lintChart() in single-value context
cmd/helm/lint_test.go:33: multiple-value lintChart() in single-value context
```
cc/ @prydonius

**Comment by prydonius** [comment]
@michelleN woops - fixed!

**Comment by michelleN** [comment]
```
helm [feat/836-lint-multiple-charts*]$ bin/helm lint funcharts/*.tgz
==> Skipping funcharts/*.tgz
open funcharts/*.tgz: no such file or directory

0 chart(s) linted, no failures
```
nice

**Pull request merged**